lizmap/modules/view/controllers/media.classic.php

```
@@ -119,86 +120,121 @@ public function getMedia()
-        }
-        $n_abspath = implode('/', $n_abspath);
-        $n_abspath = str_replace('./', '', $n_abspath);
-
-        $ok = true;
-        // Only allow files within the repository for safety reasons
-        // and in the media folder
-        // accept ../media folder to centralize medias
-        $repex = explode('/', $n_repositoryPath);
-        array_pop($repex);
-        $reptest = implode('/', $repex);
-        if (!preg_match('#^'.$n_repositoryPath.'(/)?media/#', $n_abspath)
-            && !preg_match('#^'.$reptest.'(/)?media/#', $n_abspath)
-        ) {
-            $ok = false;
-        }
-
-        // Check if file exists
-        if ($ok && !is_file($abspath)) {
-            $ok = false;
-        }
-
-        // Redirect if errors
-        if (!$ok) {
-            $content = 'No media file in the specified path: '.$path;
-            if (is_link($repositoryPath.'/'.$path)) {
-                $content .= ' '.readlink($repositoryPath.'/'.$path);
+        // check if a remote file is requested from WebDAV Storage
+        if (strpos($path, 'dav/') === 0) {
+            // replace the path with the webdav url
+            $profile = RemoteStorageRequest::getProfile('webdav');
+            if ($profile) {
+                $webdavPath = str_replace('dav/', $profile['baseUri'], $path);
+                // assumes that last part of the url is the filename
+                $urlPart = explode('/', $webdavPath);
+                $fileName = array_pop($urlPart);
+                $url = implode('/', $urlPart).'/';
+
+                $tempFileUrl = RemoteStorageRequest::getRemoteFile($url, $fileName);
+                if (!is_file($tempFileUrl)) {
+                    $content = 'No media file in the specified path: '.$path;
+
+                    return $this->error404($content);
+                }
+                $isWebDavResource = true;
+                $finalPath = $tempFileUrl;
+            }
```

maybe return an error if no profile ?
```suggestion
            } else {
                 return $this->error404('Unable to get media');
            }
```